modules/apps/object/object-dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Attachment/Attachment.tsx

```diff
@@ -237,10 +243,11 @@ export default function Attachment({
 	};
 
 	return (
-		<FieldBase {...otherProps} {...error}>
+		<FieldBase {...otherProps} {...error} readOnly={readOnly}>
 			<div className="inline-item lfr-objects__attachment">
 				<ClayButton
 					className="lfr-objects__attachment-button"
+					disabled={readOnly}
 					displayType="secondary"
 					onClick={() => {
 						setError({});
```

Comment:
Hey @Manda-vit, from what I saw in the ticket I think that this button should not appear on the screen when `readOnly` is `true`, so we can make a condition for it very similar to what you did with the delete and download button, and you can remove the `disabled` property from the button itself:

```tsx
{!readOnly && (
	<ClayButton
		className="lfr-objects__attachment-button"
		displayType="secondary"
		onClick={() => {
			setError({});

			if (fileSource === 'documentsAndMedia') {
				openSelectionModal({
					onSelect: handleSelectedItem,
					selectEventName: `${portletNamespace}selectAttachmentEntry`,
					title: Liferay.Language.get('select-file'),
					url,
				});
			}
			else if (fileSource === 'userComputer') {
				const filePicker = inputRef.current;
				if (filePicker) {
					filePicker.value = '';
					filePicker.click();
				}
			}
		}}
	>
		{Liferay.Language.get('select-file')}
	</ClayButton>
)}
```